dj_rest_auth/app_settings.py

```diff
@@ -1,56 +1,84 @@
+    'PASSWORD_RESET_SERIALIZER': 'dj_rest_auth.serializers.PasswordResetSerializer',
+    'PASSWORD_RESET_CONFIRM_SERIALIZER': 'dj_rest_auth.serializers.PasswordResetConfirmSerializer',
+    'PASSWORD_CHANGE_SERIALIZER': 'dj_rest_auth.serializers.PasswordChangeSerializer',
 
-create_token = import_callable(getattr(settings, 'REST_AUTH_TOKEN_CREATOR', default_create_token))
+    'REGISTER_SERIALIZER': 'dj_rest_auth.registration.serializers.RegisterSerializer',
 
-serializers = getattr(settings, 'REST_AUTH_SERIALIZERS', {})
+    'REGISTER_PERMISSION_CLASSES': ('rest_framework.permissions.AllowAny',),
 
-TokenSerializer = import_callable(serializers.get('TOKEN_SERIALIZER', DefaultTokenSerializer))
+    'TOKEN_MODEL': 'rest_framework.authtoken.models.Token',
+    'TOKEN_CREATOR': 'dj_rest_auth.utils.default_create_token',
 
-JWTSerializer = import_callable(serializers.get('JWT_SERIALIZER', DefaultJWTSerializer))
+    'PASSWORD_RESET_USE_SITES_DOMAIN': False,
+    'OLD_PASSWORD_FIELD_ENABLED': False,
+    'LOGOUT_ON_PASSWORD_CHANGE': False,
+    'SESSION_LOGIN': True,
+    'USE_JWT': False,
 
-JWTSerializerWithExpiration = import_callable(serializers.get('JWT_SERIALIZER_WITH_EXPIRATION', DefaultJWTSerializerWithExpiration))
+    'JWT_AUTH_COOKIE': None,
+    'JWT_AUTH_REFRESH_COOKIE': None,
+    'JWT_AUTH_REFRESH_COOKIE_PATH': '/',
+    'JWT_AUTH_SECURE': False,
+    'JWT_AUTH_HTTPONLY': True,
+    'JWT_AUTH_SAMESITE': 'Lax',
+    'JWT_AUTH_RETURN_EXPIRATION': False,
+    'JWT_AUTH_COOKIE_USE_CSRF': False,
+    'JWT_AUTH_COOKIE_ENFORCE_CSRF_ON_UNAUTHENTICATED': False,
+}
 
-UserDetailsSerializer = import_callable(serializers.get('USER_DETAILS_SERIALIZER', DefaultUserDetailsSerializer))
+# List of settings that may be in string import notation.
+IMPORT_STRINGS = (
+    'TOKEN_CREATOR',
+    'TOKEN_MODEL',
+    'TOKEN_SERIALIZER',
+    'JWT_SERIALIZER',
+    'JWT_SERIALIZER_WITH_EXPIRATION',
+    'JWT_TOKEN_CLAIMS_SERIALIZER',
+    'USER_DETAILS_SERIALIZER',
+    'LOGIN_SERIALIZER',
+    'PASSWORD_RESET_SERIALIZER',
+    'PASSWORD_RESET_CONFIRM_SERIALIZER',
+    'PASSWORD_CHANGE_SERIALIZER',
+    'REGISTER_SERIALIZER',
+    'REGISTER_PERMISSION_CLASSES',
+)
 
-LoginSerializer = import_callable(serializers.get('LOGIN_SERIALIZER', DefaultLoginSerializer))
+# List of settings that have been removed
+REMOVED_SETTINGS = ( )
 
-PasswordResetSerializer = import_callable(
-    serializers.get(
-        'PASSWORD_RESET_SERIALIZER', DefaultPasswordResetSerializer,
-    ),
-)
 
-PasswordResetConfirmSerializer = import_callable(
-    serializers.get(
-        'PASSWORD_RESET_CONFIRM_SERIALIZER', DefaultPasswordResetConfirmSerializer,
-    ),
-)
+class APISettings(_APISettings):  # pragma: no cover
+    def __check_user_settings(self, user_settings):
+        from .utils import format_lazy
+        SETTINGS_DOC = 'https://dj-rest-auth.readthedocs.io/en/latest/configuration.html'
+
+        for setting in REMOVED_SETTINGS:
```

Comment:
Can you confirm this only gets runs once at setup time?

Reply:
I'm not sure but, i add `print("I am called")` in `__check_user_settings` then `coverage run ./runtests.py` surprisingly it's not printed. Try it with demo project and it's printed two times.